File: sample/iosSwiftApp/iosSwiftApp/WheelDockedScreen.swift
//
//  WheelDockedScreen.swift
//  Nepali Date Picker
//
//  The wheel picker and the docked field that opens a calendar popup.
//

import SwiftUI
import nepali_date_picker

struct WheelDockedScreen: View {
    @State private var wheel: CustomCalendar?
    @State private var nepaliWheel: CustomCalendar?
    @State private var docked: CustomCalendar?
    @State private var dockedShort: CustomCalendar?

    private let preselectedDate = SimpleDate(year: 2081, month: 6, dayOfMonth: 10)

    var body: some View {
        ScrollView {
            VStack(spacing: 16) {
                DemoSection(
                    title: "Wheel picker",
                    subtitle: "Scrolling year, month and day columns. Starts on today when no date is given."
                ) {
                    AutoSized(measurementHeight: 240) { report in
                        NepaliWheelDatePickerView(onHeightChange: report, initialDate: nil) { wheel = $0 }
                    }
                    SelectionSummary(selection: wheel)
                }

                DemoSection(
                    title: "Wheel, pre-selected and localized",
                    subtitle: "Opens on Ashoj 10, 2081 with Nepali labels and Devanagari digits."
                ) {
                    AutoSized(measurementHeight: 240) { report in
                        NepaliWheelDatePickerView(
                            onHeightChange: report,
                            initialDate: preselectedDate,
                            locale: SampleDefaults.nepali
                        ) { nepaliWheel = $0 }
                    }
                    SelectionSummary(selection: nepaliWheel)
                }

                DemoSection(
                    title: "Docked picker",
                    subtitle: "A compact field sized to its content. A Compose popup is clipped to its host, so raise measurementHeight if you want the calendar to open inline."
                ) {
                    AutoSized { report in
                        NepaliDatePickerDockedView(onHeightChange: report, initialSelectedDate: preselectedDate) { docked = $0 }
                    }
                    SelectionSummary(selection: docked)
                }

                DemoSection(
                    title: "Docked with a compact format",
                    subtitle: "The same field writing the selection in the compact year-first style."
                ) {
                    AutoSized { report in
                        NepaliDatePickerDockedView(
                            onHeightChange: report,
                            initialSelectedDate: nil,
                            dateFormatStyle: .compactYmd,
                            showTodayButton: false
                        ) { dockedShort = $0 }
                    }
                    SelectionSummary(selection: dockedShort)
                }
            }
            .padding(.horizontal, 12)
            .padding(.vertical, 16)
        }
        .navigationTitle("Wheel & Docked")
    }
}

File: sample/flutterApp/lib/src/screens/wheel_docked_screen.dart
// Copyright © 2026 Shiva Thapa (@shivathapaa). All rights reserved.
// Use of this source code is governed by the MPL 2.0 license that can be
// found in the LICENSE file.

import 'package:flutter/material.dart';
import 'package:nepali_date_picker_kmp/nepali_date_picker_kmp.dart';

import '../support/demo_section.dart';
import '../support/sample_defaults.dart';

/// The scrolling wheel and the compact docked field.
class WheelDockedScreen extends StatefulWidget {
  const WheelDockedScreen({super.key});

  @override
  State<WheelDockedScreen> createState() => _WheelDockedScreenState();
}

class _WheelDockedScreenState extends State<WheelDockedScreen> {
  static const _preselected = SimpleDate(2081, 6, 10);

  String _wheel = 'none';
  String _nepaliWheel = 'none';
  String _docked = 'none';
  String _compactDocked = 'none';

  @override
  Widget build(BuildContext context) => DemoScreen(
        title: 'Wheel & Docked',
        children: [
          DemoSection(
            title: 'Wheel picker',
            subtitle: 'Starts on today; every spin reports a full calendar.',
            child: Column(children: [
              NepaliWheelDatePicker(
                onDateChange: (date) =>
                    setState(() => _wheel = calendarText(date)),
              ),
              SelectionSummary(_wheel),
            ]),
          ),
          DemoSection(
            title: 'Wheel, pre-selected and localized',
            subtitle: 'Opens on Ashoj 10, 2081 with Nepali names and digits.',
            child: Column(children: [
              NepaliWheelDatePicker(
                initialDate: _preselected,
                locale: SampleDefaults.nepali,
                onDateChange: (date) =>
                    setState(() => _nepaliWheel = calendarText(date)),
              ),
              SelectionSummary(_nepaliWheel),
            ]),
          ),
          DemoSection(
            title: 'Docked picker',
            subtitle: 'A text field with a dropdown calendar under it.',
            child: Column(children: [
              NepaliDatePickerDocked(
                label: 'Date',
                initialSelectedDate: _preselected,
                onDateSelected: (date) =>
                    setState(() => _docked = calendarText(date)),
              ),
              SelectionSummary(_docked),
            ]),
          ),
          DemoSection(
            title: 'Docked with a compact format',
            subtitle: 'The field writes a compact year-first date and drops '
                'the today button.',
            child: Column(children: [
              NepaliDatePickerDocked(
                label: 'Date',
                initialSelectedDate: _preselected,
                dateFormatStyle: NepaliDateFormatStyle.compactYmd,
                showTodayButton: false,
                onDateSelected: (date) =>
                    setState(() => _compactDocked = calendarText(date)),
              ),
              SelectionSummary(_compactDocked),
            ]),
          ),
        ],
      );
}
